src/runtime/components/svg.ts

```diff
@@ -36,10 +40,19 @@ export const NuxtIconSvg = /* @__PURE__ */ defineComponent({
       }
     }
 
+    const updateStrokeWidth = (el) => {
+      if (props.stroke === null) return
+      el.querySelectorAll('[stroke-width]').forEach((el: HTMLElement) => {
+        el.setAttribute('stroke-width', props.stroke)
+      })
```

Comment:
I don't think mutating DOM is a robust approach, also it won't work with SSR

/cc @cyberalien is that possible to have such functionality in `@iconify/vue` or even `@iconify/utils/lib/css/icon` so that can work also for CSS icons?

Reply:
This is a very specific use case and not a good solution. What about other use cases? This might end up being a mess if every problem is solved like this.

Why is it a bad solution?
- Doesn't take into account possibility of multiple stroke-width values in same file
- Doesn't handle animations
- What if there is no stroke-width (default value is `1`)?

What other use cases?
- Changing color for icons with multiple colors
- Changing opacities of semi-transparent shapes
- Changing animation duration

Adding solutions for each use case would only bloat icon component with code that will be rarely used.

I think better approach is a callback to customise icon before rendering it. I've implemented such solution in Tailwind CSS plugin, wanted to add it to icon components too, but didn't get to that yet.

Tailwind CSS plugin docs: https://iconify.design/docs/usage/css/tailwind/iconify/customise.html

That would allow customising anything with simple string replacements.

Current ideas for implementation:
1. Add a function parameter to component to customise icon. Can be used for each instance without affecting other instances.
2. Add a callback for all icons.
3. Trigger event to customise icon. Can be attached to each instance or to whole app.

Code for idea #1:
```js
function customise(content, name, prefix) {
  if (prefix === 'tabler') return content.replaceAll('stroke-width="2"', 'stroke-width="1.5"');
  return content;
}
```
```html
<Icon icon="tabler:alert" :customise="customise" />
```

Code for idea #2:
```js
import { onCustomise } from '@iconify/vue';

onCustomise((content, name, prefix) => {
  if (prefix === 'tabler') return content.replaceAll('stroke-width="2"', 'stroke-width="1.5"');
  return content;
});
```

Code for idea #3:
```js
function customise(content, name, prefix) {
  if (prefix === 'tabler') return content.replaceAll('stroke-width="2"', 'stroke-width="1.5"');
  return content;
}
```
```html
<Icon icon="tabler:alert" @iconifycustomise="customise" />
```
Not sure which solution is better or maybe add multiple solutions at once, or maybe there are better ideas?

Reply:
From my side, I think `Idea 1` should be good enough. We will provide a configuration entry to pass down to the component/usage, and hide the API details from users

Reply:
I also agree with Idea 1! It will solve the issue of being able to change the stroke-width.

Reply:
Thanks!

Published `@iconify/vue@dev` with `customise` prop.

Usage examples from test (one as function, one as inline function):
```html
<Icon icon="line-md:arrow-right" :customise="customise1" />
			<Icon
				icon="line-md:arrow-right"
				:customise="
					(content) =>
						content.replace(
							/stroke-width=&quot;2&quot;/g,
							'stroke-width=&quot;1&quot;'
						)
				"
			/>
```
```ts
function customise1(
	content: string,
	name: string,
	prefix: string,
	provider: string
) {
	if (name !== 'arrow-right' || prefix !== 'line-md' || provider !== '') {
		throw new Error('Bad params in customise callback');
	}
	return content.replace(/stroke-width="2"/g, 'stroke-width="1.5"');
}
```
Callback is called only for icons loaded from API.

If icon data is passed as prop, callback is not called because you can customise icon before passing it as prop.